ETLParser.py

```diff
@@ -0,0 +1,166 @@
+
+    def get_content(self):
+        self.pos = 0
+
+        # Skip everything up to and including the h1 element (the title).
+        while self.pos < len(self.lines):
+            line = self.lines[self.pos]
+            self.pos = self.pos + 1
+            if _h1.match(line):
+                break
+
+        # Everything up to the first h2 element is model-level content.
+        content = ''
+        while self.pos < len(self.lines):
+            line = self.lines[self.pos]
+            self.pos = self.pos + 1
+            if _h2.match(line):
+                break
+            content = content + line
+
+        self.add_content(content.strip('\n'))
+
+        # First h2 element is the table of contents. Skip it, up to the next h2
+        while self.pos < len(self.lines):
+            line = self.lines[self.pos]
+            if _h2.match(line):
+                break
+            self.pos = self.pos + 1
+
+    def get_next_table(self):
+        # self.pos is now pointing to an h2 line.
+        # Read up to the next h2 and parse table content.
+
+        # Table name line looks something like this: "## 1.1 PERSON".
+        # Parse out the actual table name.
+        table_name = re.match(r'^\#\#\s*\d+\.\d+\s*(?P<name>.*)$',
+                              self.lines[self.pos])
+        table_name = table_name.group('name').lower()
+
+        self.pos = self.pos + 1
+        if self.lines[self.pos].strip() == '':
+            self.pos = self.pos + 1
+
+        # Everything after the h2 and up to the line
+        # 'Field |Required | Data Type | Description | PEDSnet Conventions'
+        # is table content.
+        content = ''
+        while self.pos < len(self.lines):
+            line = self.lines[self.pos]
+            self.pos = self.pos + 1
+
+            if re.match(r'^Field\s*\|\s*Required\s*\|', line):
+                # skip the '--- | --- | --- | --- | ---' line as well
+                self.pos = self.pos + 1
+                break
+
+            content = content + line
+
+        self.add_table(table_name)
+
+        # Read field information.
+        # The format is
+        # 'Field |Required | Data Type | Description | PEDSnet Conventions'
+        while self.pos < len(self.lines):
+            line = self.lines[self.pos]
+
+            if _h2.match(line):
+                break
+
+            field = re.match(r'^(?P<name>[^\|]*)\|(?P<req>[^\|]*)\|'
+                             r'(?P<type>[^\|]*)\|(?P<desc>[^\|]*)(\|'
+                             r'(?P<conv>[^\|]*))?\|?$',
+                             line)
+
+            if field:
+                name = field.group('name').lower().strip()
+                req = field.group('req').lower().strip().startswith('yes')
+                data_type = field.group('type').strip()
+                desc = field.group('desc').strip()
+                conv = field.group('conv')
+                if conv:
+                    conv = conv.strip()
```

Comment:
if ETL conventions aren't specified, then the field is shortened to 'Field |Required | Data Type | Description', usually without a training '|', and so field.group('conv') ends up being not set. 
So I get the error 
'NoneType' object has no attribute 'strip'
unless I do check for this.